packages/ripple/src/index.ts

```diff
@@ -0,0 +1,119 @@
+import { getRippleGlobalNamespace } from './getRippleGlobalNamespace';
+import { Options, colorMap, Mode } from './utils';
+
+const TRANSITION_TIME = 300;
+const RIPPLE_NAMESPACE = getRippleGlobalNamespace();
+
+function rippleEvent(event: MouseEvent) {
+  if (
+    RIPPLE_NAMESPACE?.registeredRippleElements.has(event.target as HTMLElement)
+  ) {
+    createRippleEffect(event);
+  }
+}
+
+export function registerRipple(node: HTMLElement, options: Options) {
+  if (!RIPPLE_NAMESPACE) {
+    return;
+  }
+
+  RIPPLE_NAMESPACE.registeredRippleElements.set(node, options);
+
+  if (!RIPPLE_NAMESPACE.setRippleListener) {
+    document.addEventListener('click', rippleEvent, { passive: true });
+
+    const styles = document.createElement('style');
+    styles.innerHTML = staticRippleStyles;
+    document.head.append(styles);
+
+    RIPPLE_NAMESPACE.setRippleListener = true;
+  }
+
+  return () => RIPPLE_NAMESPACE.registeredRippleElements.delete(node);
+}
+
+function createRippleEffect(event: MouseEvent) {
+  const target = event.target as HTMLElement | null;
+  const foundNode = RIPPLE_NAMESPACE?.registeredRippleElements.get(
```

Comment:
This is neat. Hows the performance vs the object lookup we were considering?

Reply:
Just sent you a pen offline with perf update, lmk what you think